File: AVc/app/src/main/java/com/hfad/avc/ui/database/Template.java
package com.hfad.avc.ui.database;

import androidx.annotation.NonNull;
import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.library.baseAdapters.BR;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import javax.inject.Inject;

@Entity()
public class Template extends BaseObservable {
    @PrimaryKey
    @NonNull
    private String id;
    private String textTemplate = "";
    private Boolean favorite = false;

    @Inject
    public Template() {
    }

    ;

    @Bindable
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
        notifyPropertyChanged(BR.id);
    }

    @Bindable
    public String getTextTemplate() {
        return textTemplate;
    }

    public void setTextTemplate(String textTemplate) {
        this.textTemplate = textTemplate;
        notifyPropertyChanged(BR.textTemplate);
    }

    @Bindable
    public Boolean getFavorite() {
        return favorite;
    }

    public void setFavorite(Boolean favorite) {
        this.favorite = favorite;
        notifyPropertyChanged(BR.favorite);
    }

    @Override
    public String toString() {
        return "id = " + this.id +
                ", Текст: " + this.textTemplate +
                ", В избранном: " + this.favorite;
    }

}

File: AVc/app/src/main/java/com/hfad/avc/data/model/Template.kt
package com.hfad.avc.data.model

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import androidx.databinding.library.baseAdapters.BR
import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity
data class Template(
   @PrimaryKey
   private var id: String = "",
   private var textTemplate: String = "",
   private var favorite: Boolean = false
) : BaseObservable() {

   @Bindable
   fun getId(): String {
      return id
   }

   fun setId(id: String) {
      this.id = id
      notifyPropertyChanged(BR.id)
   }

   @Bindable
   fun getTextTemplate(): String {
      return textTemplate
   }

   fun setTextTemplate(textTemplate: String) {
      this.textTemplate = textTemplate
      notifyPropertyChanged(BR.textTemplate)
   }

   @Bindable
   fun getFavorite(): Boolean {
      return favorite
   }

   fun setFavorite(favorite: Boolean) {
      this.favorite = favorite
      notifyPropertyChanged(BR.favorite)
   }

   override fun toString(): String {
      return "id = " + id +
            ", Текст: " + textTemplate +
            ", В избранном: " + favorite
   }
}